document/CoreFormat.mdk

```diff
@@ -474,43 +474,47 @@ the programming language in which the programmatic construct specified by that l
 We refer to this string as a "fully qualified logical name".
 See &sect;[#location-fullyQualifiedLogicalName] for examples.
 
-Each value in the `logicalLocation` object shall be
-an array of one or more `logicalLocationComponent` objects (&sect;[#logicalLocationComponent]).
-This allows reference to results which occur in nested logical components, for example,
-a member nested within a type which is in turn nested within a namespace.
-When a result occurs in such a nested construct, the first `logicalLocationComponent` object in the array shall refer to
-the outermost construct,
-the next element shall refer to a construct nested immediately within the first construct,
-and so on, with the last element referring to the most deeply nested construct.
+Each value in the object specified by the `logicalLocations` property shall be
+a `logicalLocation` object (&sect;[#logicalLocation]).
 
-~ IsoExample
+In some cases, a logical location might be nested within another logical location
+(for example, a class nested within a namespace), referred to as its "parent."
+A logical location that is not nested within another logical location is referred to as a "top-level logical location".
+A logical location that is nested withing another logical location is referred to as a "nested logical location".
+
+If a result is detected in a nested logical location, then the `logicalLocations` object shall
+contain properties describing not only that logical location, but also properties
+describing each of its parents, up to and including the top-level logical location.
 
+~ IsoExample
+In this example, a result was detected in the C++ class `namespaceA::namespaceB::classC`. The `logicalLocations`
+object contains not only a property describing the class, but also properties describing its parents.
 ```
 "logicalLocations": {
-  "namespaceA::namespaceB::classC": [
-    {
-      "name": "namespaceA",
-      "kind": "namespace"
+    "namespaceA::namespaceB::classC": {
```

Comment:
I propose that we not require module to be at the root.

---
In reply to: [64120981](https://github.com/sarif-standard/sarif-spec/pull/213#discussion_r64120981) [](ancestors = 64120981)